rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,246 @@
+
+## Problem
+
+Dependabot was enabled across all our applications in 2018. Due to the number of applications we have and libraries we use, keeping on top of the number of PRs raised by Dependabot has been difficult.
+
+A few solutions have been implemented in the past to make this better:
+
+- [govuk-dependencies] shows the list of open PRs grouped by application, team or gem.
+- [govuk-dependencies] has a Slack bot which automatically messages in each team's channel with a list of open PRs.
+- We [allowed merging Dependabot PRs without requiring two approvals][rfc-103].
+- [Sharing the responsibility for Dependabot PRs across multiple times][dependencies-team].
+
+[govuk-dependencies]: http://govuk-dependencies.herokuapp.com/
+[rfc-103]: https://github.com/alphagov/govuk-rfcs/blob/master/rfc-103-merge-dependabot-pull-requests-with-a-single-review.md
+[dependencies-team]: https://github.com/alphagov/govuk-developer-docs/pull/2168
+
+Although these have made improvements to the process, the number of open PRs remains unmanageable. At the time of writing, there are 135 open PRs, of which there are [8 labelled as security fixes][security-prs], the oldest being 21 days old. Dependabot is currently configured to have at most 5 open PRs per repo, which means this number is not the true number and often new PRs will open the moment existing ones are closed down.
+
+[security-prs]: https://github.com/pulls?q=is%3Apr+author%3Aapp%2Fdependabot-preview+user%3Aalphagov+archived%3Afalse+is%3Aopen+label%3Asecurity
+
+This high number of open PRs is felt particularly hard by the Platform Health team, as they currently [own 55 out of 73 applications][application-owners].
+
+[application-owners]: https://docs.publishing.service.gov.uk/apps/by-team.html
+
+By having so many open PRs, it becomes difficult to be able to prioritise important updates (bug fixes, security fixes) over new versions which simply adds new features (which our apps don't necessarily use anyway). This leads to important updates being left for long periods of time and our desire to close down all the PRs means developers spend time investigating and fixing complex problems without any benefit.
+
+### Statistics
+
+I pulled together [some statistics on our Dependabot usage][dependabot-stats]:
+
+The majority of PRs are merged within a few days (on average 2.5 days). However, the PRs that require manual intervention (for example, failing tests) tend to take a significant amount of time to be merged (on order of months). Because of this talking in terms of averages tends to be less representative of the actual situation, but it can be useful to make comparisons.
+
+The libraries that take the longest time to merge are `acts-as-taggable-on`, `paper_trail`, `rails-controller-testing`, `mini_magick` and `gds-sso`. Our own libraries are merged in on average after 1.5 days, whereas third party libraries are merged in after 4 days.
```

Comment:
What's the conclusion here?

Reply:
I've reworded this section in 7826c68 which I think covers this better now!

Reply:
Um, I should have asked: "How does this influence the RFC?".

Perhaps we could delete the original first paragraph, as it doesn't seem to have an effect on the outcome - we're willing to spend the time if it's worth it.

Reply:
Yeah, that's fair. I've removed the paragraph in fb3b211f1ecdd8942e20e0c7bf2d3bd808603cc0.